### ISSUE TITLE
How to apply refresh token.

### ISSUE THREAD

**Issue by nimatrazmjo** [ItsDizzy/angular2-auth#8]
Is there any way to apply refresh token to every request? how can I apply this. if its not exist yet. Please tell where I can add it, and I will make pull request here.

Thanks

**Comment by paulhobbel** [comment]
I was planning on writing a function that automatically asks the api for a new token

**Comment by paulhobbel** [comment]
Working on it right now

**Comment by nimatrazmjo** [comment]
Cool, I will be waiting.

**Comment by paulhobbel** [comment]
@nimatullah I can't get it to work properly, check the AuthHttp and check if you can make it work correctly.

**Comment by nimatrazmjo** [comment]
@ItsDizzy same here, can not get it properly.

**Comment by paulhobbel** [comment]
Ok i'll check out what i can do later.

**Comment by paulhobbel** [comment]
Got an idea, will make a commit soon

**Comment by nimatrazmjo** [comment]
@ItsDizzy any progress ?

**Comment by paulhobbel** [comment]
Yes, but my pc is currently broken so I cannot push my current work. So far the token can be refreshed when calling a function currently making it so it does it automatically

**Comment by nimatrazmjo** [comment]
Ok, I will be waiting until you push it

**Comment by paulhobbel** [comment]
@nimatullah I pretty much gave up
